### PR TITLE
[Rule Tuning] Add Investigation Guides, Config/Logging Policy to PowerShell merged rules

### PR DESCRIPTION
**Pull request by w0rk3r**
## Summary

Adds Investigation Guides, Config/Logging Policy to the following PowerShell Script Block Logging Rules:

* PowerShell MiniDump Script
* PowerShell Suspicious Discovery Related Windows API Functions
* Suspicious Portable Executable Encoded in Powershell Script
* PowerShell Suspicious Script with Audio Capture Capabilities
* Potential Process Injection via PowerShell
* PowerShell Keylogging Script

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/windows/collection_posh_audio_capture.toml [hunk 1]
```diff
@@ -13,6 +13,54 @@ index = ["winlogbeat-*", "logs-windows.*"]
...
+- Potential Process Injection via PowerShell - 2e29e96a-b67c-455a-afe4-de6183431d0d
+
+### Response and Remediation
+
+- Immediate response should be taken to validate activity, investigate and potentially isolate activity to prevent further
+post-compromise behavior.
+
+## Config
+
+The 'PowerShell Script Block Logging' logging policy is required be configured (Enable).
```
Comment (Samirbous) [inline rules/windows/collection_posh_audio_capture.toml:48]:
```suggestion
The 'PowerShell Script Block Logging' logging policy must be enabled.
```

File: rules/windows/collection_posh_audio_capture.toml [hunk 2]
```diff
@@ -13,6 +13,54 @@ index = ["winlogbeat-*", "logs-windows.*"]
...
+
+Attackers can use PowerShell to interact with the Windows API and capture audio from input devices connected to the
+computer.
+
+#### Possible investigation steps:
+
+- Check for additional PowerShell logs that indicate that the script/command was run.
+- Inspect the script for additional capabilities, like writing the captured audio to a file, exfiltration, etc.
+- If the action is suspicious for the user, check for any other activities done by the user in the last 48 hours.
+- Look for additional alerts involving the host and the user.
```
Comment (Samirbous) [inline rules/windows/collection_posh_audio_capture.toml:30]:
```suggestion
- Examine script content that triggered the detection. 
- Investigate script execution chain (parent process tree)
- Inspect any file or network events from the suspicious powershell host process instance.
- If the action is suspicious for the user, check for any other activities done by the user in the last 48 hours.
```

File: rules/windows/credential_access_posh_minidump.toml [hunk 3]
```diff
@@ -15,9 +15,59 @@ index = ["winlogbeat-*", "logs-windows.*"]
...
+Process Memory Dump capabilities can be abused by attackers to extract credentials from LSASS or to obtain other privileged
+information stored in the process memory.
+
+#### Possible investigation steps:
+
+- Check for additional PowerShell logs that indicate that the script/command was run.
+- Gather the script content and check if it has the process that is targeted hardcoded.
+- Inspect the script for additional capabilities, like parsing of the memory dump, exfiltration, etc.
+- If the action is suspicious for the user, check for any other activities done by the user in the last 48 hours.
+- Look for additional alerts involving the host and the user.
```
Comment (Samirbous) [inline rules/windows/credential_access_posh_minidump.toml:33]:
```suggestion
- Examine script content that triggered the detection. 
- Investigate script execution chain (parent process tree)
- Inspect any file or network events from the suspicious powershell host process instance.
- If the action is suspicious for the user, check for any other activities done by the user in the last 48 hours.
```

File: rules/windows/credential_access_posh_minidump.toml [hunk 4]
```diff
@@ -15,9 +15,59 @@ index = ["winlogbeat-*", "logs-windows.*"]
...
+- Potential Process Injection via PowerShell - 2e29e96a-b67c-455a-afe4-de6183431d0d
+
+### Response and Remediation
+
+- Immediate response should be taken to validate activity, investigate and potentially isolate activity to prevent further
+post-compromise behavior.
+
+## Config
+
+The 'PowerShell Script Block Logging' logging policy is required be configured (Enable).
```
Comment (Samirbous) [inline rules/windows/credential_access_posh_minidump.toml:51]:
```suggestion
The 'PowerShell Script Block Logging' logging policy must be enabled.
```

File: rules/windows/defense_evasion_posh_process_injection.toml [hunk 5]
```diff
@@ -15,6 +15,57 @@ index = ["winlogbeat-*", "logs-windows.*"]
...
+- PowerShell PSReflect Script - 56f2e9b5-4803-4e44-a0a4-a52dc79d57fe
+
+### Response and Remediation
+
+- Immediate response should be taken to validate, investigate, and potentially contain the activity to prevent further
+post-compromise behavior.
+
+## Config
+
+The 'PowerShell Script Block Logging' logging policy is required be configured (Enable).
```
Comment (Samirbous) [inline rules/windows/defense_evasion_posh_process_injection.toml:53]:
```suggestion
The 'PowerShell Script Block Logging' logging policy must be enabled.
```

File: rules/windows/discovery_posh_suspicious_api_functions.toml [hunk 6]
```diff
@@ -16,8 +16,56 @@ index = ["winlogbeat-*", "logs-windows.*"]
...
+- PowerShell PSReflect Script - 56f2e9b5-4803-4e44-a0a4-a52dc79d57fe
+
+### Response and Remediation
+
+- Immediate response should be taken to validate activity, investigate and potentially isolate activity to prevent further
+post-compromise behavior.
+
+## Config
+
+The 'PowerShell Script Block Logging' logging policy is required be configured (Enable).
```
Comment (Samirbous) [inline rules/windows/discovery_posh_suspicious_api_functions.toml:50]:
```suggestion
The 'PowerShell Script Block Logging' logging policy must be enabled.
```

File: rules/windows/execution_posh_portable_executable.toml [hunk 7]
```diff
@@ -15,6 +15,58 @@ index = ["winlogbeat-*", "logs-windows.*"]
...
+- PowerShell PSReflect Script - 56f2e9b5-4803-4e44-a0a4-a52dc79d57fe
+
+### Response and Remediation
+
+- Immediate response should be taken to validate activity, investigate and potentially isolate activity to prevent further
+post-compromise behavior.
+
+## Config
+
+The 'PowerShell Script Block Logging' logging policy is required be configured (Enable).
```
Comment (Samirbous) [inline rules/windows/execution_posh_portable_executable.toml:51]:
```suggestion
The 'PowerShell Script Block Logging' logging policy must be enabled.
```

File: rules/windows/defense_evasion_posh_process_injection.toml [hunk 8]
```diff
@@ -15,6 +15,57 @@ index = ["winlogbeat-*", "logs-windows.*"]
...
+Red Team tooling and Malware Developers take advantage of these capabilities to develop stagers and loaders that inject
+payloads directly into the memory, without touching the disk.
+
+#### Possible investigation steps:
+
+- Check for additional PowerShell logs that indicate that the script/command was run.
+- Gather the script content and check for any payload like encoded executables or DLLs.
+- Investigate if the possible payload was executed
+- If the action is suspicious for the user, check for any other activities done by the user in the last 24 hours.
+- Look for additional alerts involving the host and the user.
```
Comment (Samirbous) [inline rules/windows/defense_evasion_posh_process_injection.toml:36]:
```suggestion
- Examine script content that triggered the detection. 
- Investigate script execution chain (parent process tree)
- Inspect any file or network events from the suspicious powershell host process instance.
- If the action is suspicious for the user, check for any other activities done by the user in the last 48 hours.
```

File: rules/windows/discovery_posh_suspicious_api_functions.toml [hunk 9]
```diff
@@ -16,8 +16,56 @@ index = ["winlogbeat-*", "logs-windows.*"]
...
+
+Attackers can use PowerShell to interact with the Win32 API to bypass file based AntiVirus detections, using libraries
+like PSReflect or Get-ProcAddress Cmdlet.
+
+#### Possible investigation steps:
+
+- Check for additional PowerShell logs that indicate that the script/command was run.
+- Gather the script content that may be split into multiple script blocks, and identify its capabilities.
+- If the action is suspicious for the user, check for any other activities done by the user in the last 48 hours.
+- Look for additional alerts involving the host and the user.
```
Comment (Samirbous) [inline rules/windows/discovery_posh_suspicious_api_functions.toml:33]:
```suggestion
- Examine script content that triggered the detection. 
- Investigate script execution chain (parent process tree).
- Inspect any file or network events from the suspicious powershell host process instance.
- If the action is suspicious for the user, check for any other activities done by the user in the last 48 hours.
```

File: rules/windows/execution_posh_portable_executable.toml [hunk 10]
```diff
@@ -15,6 +15,58 @@ index = ["winlogbeat-*", "logs-windows.*"]
...
+
+Attackers can abuse PowerShell In-Memory capabilities to inject executables into memory without touching the disk, bypassing
+AntiVirus software. These executables are generally base64 encoded.
+
+#### Possible investigation steps:
+
+- Check for additional PowerShell logs that indicate that the script/command was run.
+- Gather the script content that may be split into multiple script blocks, and identify its capabilities.
+- If the action is suspicious for the user, check for any other activities done by the user in the last 48 hours.
+- Look for additional alerts involving the host and the user.
```
Comment (Samirbous) [inline rules/windows/execution_posh_portable_executable.toml:32]:
```suggestion
- Examine script content that triggered the detection. 
- Investigate script execution chain (parent process tree).
- Inspect any file or network events from the suspicious powershell host process instance.
- If the action is suspicious for the user, check for any other activities done by the user in the last 48 hours.
```

**Review by Samirbous — APPROVED** [review]
nice work! left some optional suggestions (feel free to keep as is or not).

File: rules/windows/collection_posh_audio_capture.toml [hunk 11]
```diff
@@ -13,6 +13,54 @@ index = ["winlogbeat-*", "logs-windows.*"]
 language = "kuery"
 license = "Elastic License v2"
 name = "PowerShell Suspicious Script with Audio Capture Capabilities"
+note = """## Triage and analysis.
+
+### Investigating PowerShell Suspicious Script with Audio Capture Capabilities
+
+PowerShell is one of the main tools in the belt of system administrators for automation, report routines, and other tasks.
```
Comment (brokensound77) [inline rules/windows/collection_posh_audio_capture.toml:20]:
```suggestion
PowerShell is one of the main tools used by system administrators for automation, report routines, and other tasks.
```

File: rules/windows/collection_posh_keylogger.toml [hunk 12]
```diff
@@ -14,6 +14,53 @@ index = ["winlogbeat-*", "logs-windows.*"]
 language = "kuery"
 license = "Elastic License v2"
 name = "PowerShell Keylogging Script"
+note = """## Triage and analysis.
+
+### Investigating PowerShell Keylogging Script
+
+PowerShell is one of the main tools in the belt of system administrators for automation, report routines, and other tasks.
```
Comment (brokensound77) [inline rules/windows/collection_posh_keylogger.toml:21]:
```suggestion
PowerShell is one of the main tools used by system administrators for automation, report routines, and other tasks.
```

File: rules/windows/credential_access_posh_minidump.toml [hunk 13]
```diff
@@ -15,9 +15,58 @@ index = ["winlogbeat-*", "logs-windows.*"]
 language = "kuery"
 license = "Elastic License v2"
 name = "PowerShell MiniDump Script"
+note = """## Triage and analysis.
+
+### Investigating PowerShell MiniDump Script
+
+PowerShell is one of the main tools in the belt of system administrators for automation, report routines, and other tasks.
```
Comment (brokensound77) [inline rules/windows/credential_access_posh_minidump.toml:22]:
```suggestion
PowerShell is one of the main tools used by system administrators for automation, report routines, and other tasks.
```

File: rules/windows/defense_evasion_posh_process_injection.toml [hunk 14]
```diff
@@ -15,6 +15,56 @@ index = ["winlogbeat-*", "logs-windows.*"]
 language = "kuery"
 license = "Elastic License v2"
 name = "Potential Process Injection via PowerShell"
+note = """## Triage and analysis.
+
+### Investigating Potential Process Injection via PowerShell
+
+PowerShell is one of the main tools in the belt of system administrators for automation, report routines, and other tasks.
```
Comment (brokensound77) [inline rules/windows/defense_evasion_posh_process_injection.toml:22]:
```suggestion
PowerShell is one of the main tools used by system administrators for automation, report routines, and other tasks.
```

File: rules/windows/discovery_posh_suspicious_api_functions.toml [hunk 15]
```diff
@@ -16,8 +16,56 @@ index = ["winlogbeat-*", "logs-windows.*"]
 language = "kuery"
 license = "Elastic License v2"
 name = "PowerShell Suspicious Discovery Related Windows API Functions"
+note = """## Triage and analysis.
+
+### Investigating PowerShell Suspicious Discovery Related Windows API Functions
+
+PowerShell is one of the main tools in the belt of system administrators for automation, report routines, and other tasks.
```
Comment (brokensound77) [inline rules/windows/discovery_posh_suspicious_api_functions.toml:23]:
```suggestion
PowerShell is one of the main tools used by system administrators for automation, report routines, and other tasks.
```

File: rules/windows/execution_posh_portable_executable.toml [hunk 16]
```diff
@@ -15,6 +15,58 @@ index = ["winlogbeat-*", "logs-windows.*"]
 language = "kuery"
 license = "Elastic License v2"
 name = "Suspicious Portable Executable Encoded in Powershell Script"
+note = """## Triage and analysis.
+
+### Investigating Suspicious Portable Executable Encoded in Powershell Script
+
+PowerShell is one of the main tools in the belt of system administrators for automation, report routines, and other tasks.
```
Comment (brokensound77) [inline rules/windows/execution_posh_portable_executable.toml:22]:
```suggestion
PowerShell is one of the main tools used by system administrators for automation, report routines, and other tasks.
```

**Review by brokensound77 — APPROVED** [review]
✔️

**Pull request merged**